### ISSUE TITLE
Micronaut dockerBuildNative does not build aarch64 artifact

### ISSUE THREAD

**Issue by CezaryBD** [micronaut-projects/micronaut-gradle-plugin#829]
### Expected Behavior

Hello,

I have tried basic example from Micronaut Launch with 0 added features.
My machine is MacBook Pro M1.

According to documentation:

The plugin will detect the host operating system architecture (based on the os.arch Java system property) and will install the corresponding GraalVM binary distribution inside the Docker image. This means that when running packaging from an X86_64 (Intel/AMD) machine, the produced native executable will be an amd64 binary, whilst on an ARM host (such as the new Mac M1) it will be an aarch64 binary.
https://micronaut-projects.github.io/micronaut-gradle-plugin/latest/#_architecture_considerations

### Actual Behaviour

I can still see that it builds amd64 artifact and I cannot run it on my machine:

docker image inspect --format "{{.ID}} {{.RepoTags}} {{.Architecture}}" demo
sha256:08b3dd3b4eca5a81666c326dc7e9edfe919171f031d04b98f0efc23fd77be3d9 [demo:latest] amd64

### Steps To Reproduce

1. Use https://micronaut.io/launch/
2. Version 4.1.1, Java, Gradle, Junit - no added features
3. Make sure your machine is a Mac with M1/M2 processor
4. Run dockerBuildNative
5. Inspect image via CLI or via Docker Desktop

### Environment Information

MacOS Ventura 13.5.2

openjdk 17.0.8 2023-07-18 LTS
OpenJDK Runtime Environment Corretto-17.0.8.7.1 (build 17.0.8+7-LTS)
OpenJDK 64-Bit Server VM Corretto-17.0.8.7.1 (build 17.0.8+7-LTS, mixed mode, sharing)

### Version

4.1.1

**Comment by timyates** [comment]
TL;DR: as a workaround, you need to change

So the native image is built correctly as an aarch64 image.

The issue then is that we copy this image into an image `FROM frolvlad/alpine-glibc:alpine-3.16` and there's no aarch64 version of alpine linux...

As a workaround you need to change the base image with something like (this is not a recommendation of base image) 😉:

```
tasks.named('dockerfileNative') {
    baseImage('cgr.dev/chainguard/wolfi-base:latest')
}
```

Which has an ARM version, so will generate a final ARM image

```
❯ docker image inspect --format "{{.ID}} {{.RepoTags}} {{.Architecture}}" demo
sha256:cf0249cf13db72ecd66465a0498a3fc45fd12010b816e059f2d6a741ac4af04f [demo:latest] arm64
```

We need to think about how to handle this in the plugin, and at least document this while we think of a base image to default to...

**Comment by CezaryBD** [comment]
> TL;DR: as a workaround, you need to change the base image to one with an ARM variant
> 
> So the native image is built correctly as an aarch64 image.
> 
> The issue then is that we copy this image into an image `FROM frolvlad/alpine-glibc:alpine-3.16` and there's no aarch64 version of alpine linux...
> 
> As a workaround you need to change the base image with something like (this is not a recommendation of base image) 😉:
> 
> ```
> tasks.named('dockerfileNative') {
>     baseImage('cgr.dev/chainguard/wolfi-base:latest')
> }
> ```
> 
> Which has an ARM version, so will generate a final ARM image
> 
> ```
> ❯ docker image inspect --format "{{.ID}} {{.RepoTags}} {{.Architecture}}" demo
> sha256:cf0249cf13db72ecd66465a0498a3fc45fd12010b816e059f2d6a741ac4af04f [demo:latest] arm64
> ```
> 
> We need to think about how to handle this in the plugin, and at least document this while we think of a base image to default to...

Yes, correct.
Adding e.g. `
tasks.named('dockerfileNative') {
    baseImage('arm64v8/ubuntu:latest')
}
`
to **build.gradle** solves the issue.

Are there any candidates for a fallback of baseImage you are currently considering?

**Comment by timyates** [comment]
I assume alpine was chose due to it's size, so I am thinking about using wolfi-base as per the comment above (I believe it's only 11MB)

Do you have any better images that come to mind?

**Comment by CezaryBD** [comment]
Not really, this one is pretty small and should work. I dunno about future support of this distribution, but as long it is actively maintained, I guess no issue.
